java/core/src/test/java/sleeper/core/iterator/AggregationIteratorImplTest.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.iterator;
+
+import org.junit.jupiter.api.Test;
+
+import sleeper.core.iterator.AggregationFilteringIterator.Aggregation;
+import sleeper.core.iterator.AggregationFilteringIterator.AggregationOp;
+import sleeper.core.iterator.AggregationFilteringIterator.FilterAggregationConfig;
+import sleeper.core.record.Record;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static sleeper.core.iterator.AggregatorIteratorImpl.aggregateOnTo;
+
+public class AggregationIteratorImplTest {
+
+    private static FilterAggregationConfig createConfig() {
+        return new FilterAggregationConfig(List.of("key1", "key2", "sort_key", "sort_key2"), Optional.empty(),
+                0,
+                List.of());
+    }
+
+    private static FilterAggregationConfig createConfigWithAggregations() {
+        return new FilterAggregationConfig(List.of("key1", "key2", "sort_key", "sort_key2"), Optional.empty(),
+                0,
+                List.of(new Aggregation("value1", AggregationOp.SUM),
+                        new Aggregation("value2", AggregationOp.MIN)));
+    }
+
+    private static void checkAggregationResult(List<Record> records, List<Record> expectedResults) throws Exception {
+        List<Record> results = aggregate(records);
+        assertThat(results).containsExactlyElementsOf(expectedResults);
```

Comment:
I think this would be easier to read if we make the assertion explicit. In each of the tests, we can replace:

```java
checkAggregationResult(records, expected);
```

With:

```java
assertThat(aggregate(records)).containsExactlyElementsOf(expected);
```